bigbluebutton-client/src/org/bigbluebutton/modules/whiteboard/views/TextDrawListener.as

```diff
@@ -16,122 +16,136 @@
+
+        private var feedback:RectangleFeedbackTextBox = new RectangleFeedbackTextBox();
+
+        private var _wbModel:WhiteboardModel;
+
+        public function TextDrawListener(idGenerator:AnnotationIDGenerator, wbCanvas:WhiteboardCanvas, sendShapeFrequency:int, shapeFactory:ShapeFactory, wbModel:WhiteboardModel) {
+            _idGenerator = idGenerator;
+            _wbCanvas = wbCanvas;
+            _sendFrequency = sendShapeFrequency;
+            _shapeFactory = shapeFactory;
+            _wbModel = wbModel;
         }
 
-                feedback.draw(_mouseXDown, _mouseYDown, mouseX - _mouseXDown, mouseY - _mouseYDown);
-                _wbCanvas.addRawChild(feedback);                
+        public function ctrlKeyDown(down:Boolean):void {
+            // Ignore
+        }
+
+        public function onMouseDown(mouseX:Number, mouseY:Number, tool:WhiteboardTool):void {
+            if (tool.graphicType == WhiteboardConstants.TYPE_TEXT) {
+                _mouseXDown = _mouseXMove = mouseX;
+                _mouseYDown = _mouseYMove = mouseY;
+
+                // We have to keep track if the user has pressed the mouse. A mouseup event is
+                // dispatched when the mouse goes out of the canvas, theu we end up sending a new text
+                // even if the user has mousedDown yet.
+                _mousedDown = true;
             }
+        }
+
+        public function onMouseMove(mouseX:Number, mouseY:Number, tool:WhiteboardTool):void {
+            if (tool.graphicType == WhiteboardConstants.TYPE_TEXT && _mousedDown) {
+                _mouseXMove = mouseX;
+                _mouseYMove = mouseY;
+
+                if (_wbCanvas.contains(feedback)) {
+                    _wbCanvas.removeRawChild(feedback);
+                }
 
+                feedback.draw(_mouseXDown, _mouseYDown, mouseX - _mouseXDown, mouseY - _mouseYDown);
+                _wbCanvas.addRawChild(feedback);
+            }
         }
-        
-        public function onMouseUp(mouseX:Number, mouseY:Number, tool:WhiteboardTool):void
-        {
+
+        public function onMouseUp(mouseX:Number, mouseY:Number, tool:WhiteboardTool):void {
             // We have to use the saved X and Y from the last mouse move rather than the X and 
             // Y from MouseUp because the latter might be outside the bounds of the area.
-            
+
             if (tool.graphicType == WhiteboardConstants.TYPE_TEXT && _mousedDown) {
                 feedback.clear();
                 if (_wbCanvas.contains(feedback)) {
                     _wbCanvas.removeRawChild(feedback);
                 }
-                
+
                 _mousedDown = false;
 
                 var tbWidth:Number = Math.abs(_mouseXMove - _mouseXDown);
                 var tbHeight:Number = Math.abs(_mouseYMove - _mouseYDown);
-                
-                if (tbHeight < 15 || tbWidth < 50) return;
-                
+
+                if (tbHeight == 0 && tbWidth == 0) {

```

Comment:
Is it accurate that ln 112-121 are the only actual changes? The rest looks like the exact same code.